src/ast/effects.h

```diff
@@ -204,6 +204,24 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
     isAtomic = true;
     if (!ignoreImplicitTraps) implicitTrap = true;
   }
+  void visitAtomicWait(AtomicWait* curr) {
+    readsMemory = true;
+    // AtomicWait doesn't strictly write memory, but it does modify the waiters
+    // list associated with the specified address, which we can think of as a
+    // write.
+    writesMemory = true;
+    isAtomic = true;
+    if (!ignoreImplicitTraps) implicitTrap = true;
```

Comment:
Wait and wake take addresses in linear memory (like using the memory address for the value of a mutex), so they can trap for the same reasons as other memory operations. See https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#wait-and-wake-operators